### PR TITLE
Move SearchService to new API endpoint

### PR DESCRIPTION
**Pull request by Ullmie02**
**Changes**
Migrate SearchService.cs from MediaBrowser.Api to Jellyfin.Api to use the new ASP.NET WebApi endpoint.

**Issues**
Part of #2872

### CODE REVIEW

File: Jellyfin.Api/BaseJellyfinApiController.cs [hunk 1]
```diff
@@ -9,5 +10,23 @@ namespace Jellyfin.Api
...
     public class BaseJellyfinApiController : ControllerBase
     {
+        /// <summary>
+        /// Splits a string at a seperating character into an array of substrings.
+        /// </summary>
+        /// <param name="value">The string to split.</param>
+        /// <param name="separator">The char that seperates the substrings.</param>
+        /// <param name="removeEmpty">Option to remove empty substrings from the array.</param>
+        /// <returns>An array of the substrings.</returns>
+        internal static string[] Split(string value, char separator, bool removeEmpty)
```
Comment (Ullmie02) [inline Jellyfin.Api/BaseJellyfinApiController.cs:20]: I migrated this method from `MediaBrowser.Api.ApiEntryPoint` but am uncertain if this is the right place to migrate it to.
As this method is used in multiple classes, I decided against a private method in `SearchController.cs`

File: Jellyfin.Api/Controllers/SearchController.cs [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Jellyfin.Api.Controllers
+{
+    /// <summary>
+    /// Search controller.
+    /// </summary>
+    [Route("/Search/Hints")]
+    [Authenticated]
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/SearchController.cs:25]:
```suggestion
    [Authorize]
```

File: Jellyfin.Api/Controllers/SearchController.cs [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+                IsNews = isNews,
+                IsSeries = isSeries,
+                IsSports = isSports
+            });
+
+            return Ok(new SearchHintResult
+            {
+                TotalRecordCount = result.TotalRecordCount,
+                SearchHints = result.Items.Select(GetSearchHintResult).ToArray()
+            });
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/SearchController.cs:124]:
```suggestion
            return new SearchHintResult
            {
                TotalRecordCount = result.TotalRecordCount,
                SearchHints = result.Items.Select(GetSearchHintResult).ToArray()
            };
```

File: Jellyfin.Api/Controllers/SearchController.cs [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+            _dtoService = dtoService;
+            _imageProcessor = imageProcessor;
+        }
+
+        /// <summary>
+        /// Gets the search hint result.
+        /// </summary>
+        /// <param name="startIndex">The record index to start at. All items with a lower index will be dropped from the results.</param>
+        /// <param name="limit">The maximum number of records to return.</param>
+        /// <param name="userId">Supply a user id to search within a user's library or omit to search all.</param>
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/SearchController.cs:57]:
```suggestion
        /// <param name="startIndex">Optional. The record index to start at. All items with a lower index will be dropped from the results.</param>
        /// <param name="limit">Optional. The maximum number of records to return.</param>
        /// <param name="userId">Optional. Supply a user id to search within a user's library or omit to search all.</param>
```

File: Jellyfin.Api/BaseJellyfinApiController.cs (hunk 1 above)
Comment (crobibero) [reply Jellyfin.Api/BaseJellyfinApiController.cs:20]: I would move it to a Helpers file, like in #2967

Comment (Ullmie02) [reply Jellyfin.Api/BaseJellyfinApiController.cs]: I moved the method to the same class created in #2967 as creating a new class seemed to be unnecessary

File: Jellyfin.Api/Controllers/SearchController.cs [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+        public ActionResult<SearchHintResult> Get(
+            [FromQuery] int? startIndex,
+            [FromQuery] int? limit,
+            [FromQuery] Guid userId,
+            [FromQuery, Required] string searchTerm,
+            [FromQuery] bool includePeople,
+            [FromQuery] bool includeMedia,
+            [FromQuery] bool includeGenres,
+            [FromQuery] bool includeStudios,
+            [FromQuery] bool includeArtists,
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/SearchController.cs:87]:
```suggestion
            [FromQuery] bool includePeople = true,
            [FromQuery] bool includeMedia = true,
            [FromQuery] bool includeGenres = true,
            [FromQuery] bool includeStudios = true,
            [FromQuery] bool includeArtists = true,
```

These parameters have a default value set in the `GetSearchHints` constructor. You'll have to move them to the end of the list, but that won't change functionality

File: Jellyfin.Api/Helpers/RequestHelpers.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    /// </summary>
+    public static class RequestHelpers
+    {
+        /// <summary>
+        /// Splits a string at a seperating character into an array of substrings.
+        /// </summary>
+        /// <param name="value">The string to split.</param>
+        /// <param name="separator">The char that seperates the substrings.</param>
+        /// <param name="removeEmpty">Option to remove empty substrings from the array.</param>
+        /// <returns>An array of the substrings.</returns>
```
Comment (crobibero) [inline Jellyfin.Api/Helpers/RequestHelpers.cs:16]:
```suggestion
        /// <summary>
        /// Splits a string at a separating character into an array of substrings.
        /// </summary>
        /// <param name="value">The string to split.</param>
        /// <param name="separator">The char that separates the substrings.</param>
        /// <param name="removeEmpty">Option to remove empty substrings from the array.</param>
        /// <returns>An array of the substrings.</returns>
```

File: Jellyfin.Api/Controllers/SearchController.cs (hunk 5 above)
Comment (Ullmie02) [reply Jellyfin.Api/Controllers/SearchController.cs]: Thanks for the guidance

File: Jellyfin.Api/Controllers/SearchController.cs [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+        /// <param name="isSeries">Optional filter for series.</param>
+        /// <param name="isNews">Optional filter for news.</param>
+        /// <param name="isKids">Optional filter for kids.</param>
+        /// <param name="isSports">Optional filter for sports.</param>
+        /// <param name="includePeople">Optional filter whether to include people.</param>
+        /// <param name="includeMedia">Optional filter whether to include media.</param>
+        /// <param name="includeGenres">Optional filter whether to include genres.</param>
+        /// <param name="includeStudios">Optional filter whether to include studios.</param>
+        /// <param name="includeArtists">Optional filter whether to include artists.</param>
+        /// <returns>An <see cref="SearchHintResult"/> with the results of the search.</returns>
```
[the remark this answers is not in the document]
Comment (Ullmie02) [reply Jellyfin.Api/Controllers/SearchController.cs:75]: No problem. I even read the comment you mentioned. But after I moved the endpoint I somehow forgot to check if I've done everything as explained.

**Pull request merged**